File: lc_soln/0102_binary_tree_level_order_traversal/solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution
{
    public List<List<int>> LevelOrder(TreeNode root)
    {
        List<List<int>> result = [];
        LinkedList<TreeNode> q = [];
        if (root is not null)
        {
            q.AddLast(root);
        }
        while (q.Count > 0)
        {
            int n = q.Count;
            var row = new List<int>(n);
            while (n-- > 0)
            {
                TreeNode node = q.First.Value;
                q.RemoveFirst();
                row.Add(node.val);
                if (node.left is not null)
                {
                    q.AddLast(node.left);
                }
                if (node.right is not null)
                {
                    q.AddLast(node.right);
                }
            }
            result.Add(row);
        }
        return result;
    }
}

File: lc_soln/0102_binary_tree_level_order_traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        if root is None:
            return result
        q = deque([root])
        while q:
            row = []
            n = len(q)
            for _ in range(n):
                node = q.popleft()
                row.append(node.val)
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
            result.append(row)
        return result


def main(): ...


if __name__ == "__main__":
    main()
